packages/node/src/utils/substrate.test.ts

```diff
@@ -65,4 +66,22 @@ describe('substrate utils', () => {
       filterExtrinsic(block.extrinsics[2], { isSigned: false }),
     ).toBeFalsy();
   });
+
+  it('decode fail message', async () => {
+    const provider = new WsProvider(ENDPOINT_KARURA);
+    const api = await ApiPromise.create({ provider });
+
+    await expect(getBlockByHeight(api, 86614)).rejects.toThrow(
+      /Unable to decode|failed decoding|unknown type/,
```

Comment:
We want to test that it throws an error with our message and not the underlying error that we wrap

Reply:
I think both are needed.